src/modeler/AutoRest.Swagger/Validation/AvoidAnonymousTypes.cs

```diff
@@ -26,9 +34,12 @@ public class AvoidAnonymousTypes : TypedRule<SwaggerObject>
         public override string MessageTemplate => Resources.AnonymousTypesDiscouraged;
 
         /// <summary>
-        /// The severity of this message (ie, debug/info/warning/error/fatal, etc)
+        /// An <paramref name="entity"/> fails this rule if it doesn't have a reference (meaning it's defined inline)
         /// </summary>
-        public override Category Severity => Category.Info;
+        /// <param name="entity">The entity to validate</param>
+        /// <returns></returns>
+        public override bool IsValid(SwaggerObject entity) => (entity?.GetType() == typeof(Schema) &&
```

Comment:
If properties is null, the expression I suggested will evaluate to 0 and pass the if condition